guides-and-standards/standard-public-repos.md

```diff
@@ -0,0 +1,45 @@
+# Standards for public GitHub repositories at Arcadia
+
+This document describes the standards for public GitHub repositories at Arcadia. These standards are intended to ensure that the code we release is well-documented, reproducible, and easy to use. All Arcadia GitHub repositories should meet these standards prior to being made public.
+
+## Aside about repository templates
+
+We have Arcadia-specific templates for [Python](https://github.com/Arcadia-Science/python-analysis-template), [Snakemake](https://github.com/Arcadia-Science/snakemake-template), and [R](https://github.com/Arcadia-Science/r-analysis-template) repositories. It is strongly recommended to start new projects using these templates, as this will mean that your repository will automatically satisfy many of the requirements below.
+
+## Standards for organization and documentation
+
+- **Repository organization**: The repository should be well-organized with descriptive file names and a clear directory structure.
+  - Scripts, notebooks, and data files should generally be in separate directories.
+  - Directory names and filenames should not include spaces or special characters.
+  - Avoid very long scripts or notebooks. If a script or notebook is more than 500-1000 lines long, consider breaking it up into smaller scripts or functions.
```

Comment:
I am not sure if the 500-1000 line is a reasonable guide. It really depends on the language you write in etc. Maybe I'd explain that files/notebooks/scripts should have specific purposes and should be grouped around them?

Reply:
How about a hybrid approach? I've seen enough 10k line python scripts to feel that it'd be nice to give a *rough* number:
```
Avoid very long scripts or notebooks. The definition of "very long" depends on context and language, but a good rule of thumb is that all files (whether modules, scripts, and notebooks) should contain code related to one specific purpose. For typical Python and R projects, this definition *generally* corresponds to a length of around 1000 lines.
```